### PR TITLE
CENNZxSpot Get Price Audit Refactors

### PR DESCRIPTION
**Pull request by hoani**
This is the result of an audit on CENNZxSpot

## Addresses:

- Group get_<>_<>_<buy/sell>_price together logically
- Renaming price functions to clarify intention
- Remove duplication of fetching reserve balances
---

## Concerns:

- none

### CODE REVIEW

File: crml/cennzx-spot/src/lib.rs [hunk 1]
```diff
@@ -436,58 +459,51 @@ impl<T: Trait> Module<T> {
...
-		ensure!(buy_amount > Zero::zero(), Error::<T>::BuyAmountNotPositive);
-
-		let core_asset_id = Self::core_asset_id();
-		let exchange_address = T::ExchangeAddressGenerator::exchange_address_for(*asset_id);
-
-		let core_asset_reserve = <pallet_generic_asset::Module<T>>::free_balance(&core_asset_id, &exchange_address);
-		let trade_asset_reserve = <pallet_generic_asset::Module<T>>::free_balance(&asset_id, &exchange_address);
+		ensure!(
+			sell_amount > Zero::zero(),
+			Error::<T>::AssetToCoreSellAmountNotAboveZero
```
Comment (jordy25519) [inline crml/cennzx-spot/src/lib.rs:488]: Yeah we should squash all of these error types into a `CannotTradeZero`

**Review by jordy25519 — APPROVED** [review]
lgtm!

File: crml/cennzx-spot/src/lib.rs (hunk 1 above)
Comment (hoani) [reply crml/cennzx-spot/src/lib.rs:488]: Will get to it soon - I have a couple more PRs in mind for this module 😉

**Review by cowboy-bebug — APPROVED** [review]
LGTM

**Pull request merged**